### PR TITLE
Feature/add google apple signin #125

### PR DESCRIPTION
**Pull request by brunotho**
This PR implements Google OAuth sign-in with account linking for existing users. 🙂
Apple sign-in was excluded. 😔

Production setup:
Set env vars

> GOOGLE_OAUTH_CLIENT_ID

> GOOGLE_OAUTH_CLIENT_SECRET

Testing: Unit tests use mocked OAuth. Manual testing requires real Google credentials (env vars + Console setup with localhost callback).

#125

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -33,6 +33,10 @@ gem 'sassc-rails' # Use SCSS for stylesheets
 # Devise for authentication
 gem 'devise', '~> 4.8'
 
+#OAauth Providers
+gem 'omniauth-google-oauth2'
+gem 'omniauth-rails_csrf_protection'
```
Comment (emmvs) [inline Gemfile:38]: Let's add an `env.sample` file so we know in the future what env vars are needed!

File: app/views/devise/shared/_links.html.erb [hunk 2]
```diff
@@ -6,6 +6,14 @@
   <%= link_to "Sign up", new_registration_path(resource_name), class: "btn button--primary w-100 btn-secondary mt-1 d-block border-0" %><br />
 <% end %>
 
+<%- if devise_mapping.omniauthable? %>
+  <%- resource_class.omniauth_providers.each do |provider| %>
+    <%= form_for "Login", url: omniauth_authorize_path(resource_name, provider), method: :post, data: {turbo: "false"} do |f| %>
+      <%= f.submit "Sign in with Google", class: "btn button--primary w-100 btn-secondary mt-1 d-block border-0" %>
```
Comment (emmvs) [inline app/views/devise/shared/_links.html.erb:12]: It says "Sign in with Google" for both sign in & sign up! 🐥 -> Let's change it depending on what you do 💡

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    begin
+      user = User.from_omniauth(auth)
```
Comment (emmvs) [inline app/controllers/users/omniauth_callbacks_controller.rb:4]: This already fails if your `first_name` is `E.` - Add a test for this edge case, and let's continue after! 🙌🏻

**Review by emmvs — CHANGES_REQUESTED** [review]
Well done for your first PR in this project!! 🎈 Let's start with these few things before we continue! But first -> let's clone and forge the project 🫶🏻
